frontend/epfl-post-teaser/controller.php

```diff
@@ -17,26 +17,42 @@ function epfl_post_teaser_block( $attributes ) {
     $post2           = Utils::get_sanitized_attribute( $attributes, 'post2' );
     $post3           = Utils::get_sanitized_attribute( $attributes, 'post3' );
     $grayBackground  = Utils::get_sanitized_attribute( $attributes, 'grayBackground', false );
+    $onlyLastPosts   = Utils::get_sanitized_attribute( $attributes, 'onlyLastPosts', false );
+    
 
     $postsCount = 0;
     $data  = [];
 
-    if ($post1 !== '') {
-        $post1 = json_decode($post1, true);
-        $data[] = $post1["value"];
+    // if we only have to display 3 lasts posts
+    if($onlyLastPosts)
+    {
+        foreach(get_posts() as $post)
```

Comment:
Peut-être une optimisation bienvenue pour supprimer la condition break ? 
```suggestion
        foreach(get_posts(['numberposts' => 3]) as $post)
```